web/src/left_sidebar_navigation_area.ts

```diff
@@ -73,19 +73,21 @@ export function update_dom_with_unread_counts(
     }
 }
 
-// TODO: Rewrite how we handle activation of narrows when doing the redesign.
-// We don't want to adjust class for all the buttons when switching narrows.
-
-function remove($elem: JQuery): void {
+function deactivate_narrow($elem: JQuery): void {
     $elem.removeClass("active-filter active-sub-filter");
 }
 
 function deselect_top_left_corner_items(): void {
-    remove($(".top_left_all_messages"));
-    remove($(".top_left_starred_messages"));
-    remove($(".top_left_mentions"));
-    remove($(".top_left_recent_view"));
-    remove($(".top_left_inbox"));
+    const narrows_to_deactivate = [
+        ".top_left_all_messages",
+        ".top_left_starred_messages",
+        ".top_left_mentions",
+        ".top_left_recent_view",
+        ".top_left_inbox",
+    ];
+    for (const narrow of narrows_to_deactivate) {
+        deactivate_narrow($(narrow));
+    }
```

Comment:
I'm pretty sure you can do this with jQuery using a single line of code rather than a loop, and that doing so is in fact more performant.